### ISSUE TITLE
Peer evaluation

### ISSUE THREAD

**Issue by owattenmaker** [CodyMattern/Mattern_CSCI2270_FinalProject#5]
First off, nice job getting something that is this working in a week.

The first issue I have is with your documentation. You don't give users a base case to look at and expect users to know what a Summoner Name is and where to get one. Very minor, but would be good to update the readme in the "How to run" section with at least a base case for users with no knowledge of what League of Legends is.

The other issues have to do with the code:

* I'm going to assume that your summoner name doesn't have spaces, because right away I noticed that the program does not handle spaces well. I would suggest putting the input into a string stream and getting rid of all spaces in the name. 

* If you enter a summoner name, it does not check if it exists. This causes an issue when you go to view ranked champion stats and the program segmentation faults because it is trying to access part of a NULL node in your BST. At first I thought that you just didn't have a checker, but then I realized that you do have a function called ifPlayed that theoretically would check the number of games played but you never call it. I would suggest doing one of 2 things:
    * Make the BST initiate with NILL nodes so you can check if they are empty. This would also solve the issue of having a account that has ranked games in one season but not the others. You can use the test account "comitragedy2" for testing this purpose, as it only has games played in season 5.
    * Look into the api documentation because I think there should be a function built in that sees if an account is a valid account. This would save you some work. 

* If you decide to go back to the main menu, it does not delete the BST from before. If I test an account that has games played in all seasons and then an account that has games played in only one, it will still report the previous seasons for the first account. All you should have to do for this one is run the destructor right before you break out of the while loop to go back to the main menu. 

On a side note, something that would improve this program is if you used a different formula for calculating best champion rather than choosing the champion with the highest KDA. Some sort of algorithm like KDA*sessions_played would work fairly well. Could get pretty advanced with it and include average deaths and kills; up to you.

Overall cool application combining a hobby with an assignment. Using an api is hard work even harder to get to work properly.

duo?

**Comment by owattenmaker** [comment]
I also wanted to add that if you have a valid account with no ranked wins in a certain season it will segfault when you select champ stats.

**Comment by owattenmaker** [comment]
The last issue is with documentation, I think that you didn't do 6 and 8 well from the Evaluation rubric:

6. Open	the	.cpp	and	.h	files.	Check the	documentation	in	the	.cpp	file.	Do	the	
methods	have	a	comment	block that	describes	the	method’s	functionality, including	a	brief	description	of what	it does, and	pre- and	post-conditions?	If documentation	is missing or	incomplete,	enter	an	issue.

8. Does	the	driver	file	include	enough	examples	of	how	to	use	the	library	code? For	example,	if	the	library	code	has	methods	for	building,	inserting,	and	deleting	nodes from	a	binary	search	tree,	does	the	driver	file	provide examples	of	how	to	do	all of	these	things? If	you	think	more	examples	are	needed,	enter	an	issue. 


The issues are apparent throughout your code and documentation.

**Comment by CodyMattern** [comment]
Owen,

Thanks for the really detailed response. The commenting  of the code and instructions in the read me file concerned me right away. Going into this project Nick ( teammate) and myself had plans to right our own API wrapper for the Riot's API ,but ended up realized we were way in over our heads for time. Yes my summoner name has no spaces and neither does my teammates so i guess a lot of these little issues we just never had the time to check for. Ultimately I wanted to have this application do what Op.GG or LolKing already do but for a desktop application. Since we ran out of time we were only able to get limited functionality and all of it has to due with ranked stats. If your account comebacks with nothing we don't have anything checking for this currently, but this will need to be fixed. I would love to play some league with you, leave me your IGN and i will add you. We won't be able to duo, cause I'm a little to high elo but that's ok.

**Comment by owattenmaker** [comment]
Yeah no problem. It is kind of fun to see what other people can come up
with. I have been there done that with going in way over my head on
projects that I think are going to be simple. Thats why for this one I just
went with a pretty simple prime number generator that can do a couple of
other things.

Another thing I was thinking that might fix your problem of having no data
and then it seg faulting is to use hash tables because then it is pretty
easy to check if that value is empty.

The duo thing was sort of a joke but it would be cool to play together
sometime. My IGN is "An Okay Player".
